build: upgrade cockroachdb/apd to v3.2.1

otan opened the pull request:
Release note (general change, bug fix): Fixed a bug where, internally,
if we print a 0 decimal with a very low exponent we use excessive
memory. This is not possible through using the decimal type, but may be
possible through crdb_internal functions.



Epic: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110527)
<!-- Reviewable:end -->

otan committed:
build: upgrade cockroachdb/apd to v3.2.1

Release note (general change, bug fix): Fixed a bug where, internally,
if we print a 0 decimal with a very low exponent we use excessive
memory. This is not possible through using the decimal type, but may be
possible through crdb_internal functions.

otan commented:
bors r=miretskiy
